source/plugins/configuration.rst

```diff
@@ -2,4 +2,141 @@
+
+If you are unsure of what to set the value of ``sharedRoot`` to, consider the following:
+
+* If you plan on having multiple configuration files (complex plugins) in the future, set the value to ``false``.
+* If you plan on having a single configuration file (less-complex plugins), set the value to ``true``.
+
+**Example - Field using** ``@DefaultConfig``
+
+.. code-block:: java
+
+    import com.google.inject.Inject;
+    import org.spongepowered.api.service.config.DefaultConfig;
+
+    @Inject
+    @DefaultConfig(sharedRoot = false)
+    private File defaultConfig;
+
+.. warning::
+
+    When your plugin is running for the first time, the returned pathname may refer to a configuration file that does not yet exist.
+
+Creating a Getter Method
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+Creating a getter method for your plugin's default configuration file may come in useful, although doing so is optional. The remainder of this guide assumes that you create a getter method similar to the one illustrated below.
+
+.. code-block:: java
+
+    public File getDefaultConfig() {
+        return defaultConfig;
+    }
+
+Setting Configuration Values
+============================
+
+.. note::
+
+    The following examples assume that the getter method for your default configuration is named ``getDefaultConfig()``, as shown in :ref:`getting-default-config`. This may differ for you depending on what you named your getter method.
+
+Defaults
+~~~~~~~~
+
+Default configuration values are a necessity for a couple of reasons:
+
+- They are the first thing server administrators will see when first opening the configuration for your plugin.
+- The lack of default configuration values may throw exceptions, depending on how you handle things.
+
+Checking whether your plugin's configuration file exists where it is expected to is an effective method of determining if default configuration values need to be set. If your plugin's configuration file does not exist where it is expected to, then the file likely needs to be created. This is shown in the example below; the example checks whether the pathname defined by the previously-defined ``getDefaultConfig()`` exists. If it does not exist, a default configuration file is created, and values are written to it using ``withValue(String path, ConfigValue value)``.
+
+.. code-block:: java
+
+     import com.typesafe.config.ConfigValueFactory;
+     import org.spongepowered.api.util.config.ConfigFile;
+     import java.io.File;
+
+     if (!getDefaultConfig().exists()) {
+        File newConfig;
+        try {
+            getDefaultConfig().createNewFile();
+        } catch (IOException exception) {
+            getLogger().log(Level.SEVERE, "The default configuration could not be created!");
+        }
+        ConfigFile config = ConfigFile.parseFile(getDefaultConfig());
+        config.withValue("plugin.version", ConfigValueFactory.fromAnyRef(1));
```

Comment:
This doesn't work because config methods are immutable. Consider them to be like a builder on the same object.

Instead, something like
```java
ConfigFile config = ConfigFile.parseFile(getDefaultConfig())
    .withValue("plugin.version", ConfigValueFactory.fromAnyRef(1));
    .withValue("plugin.doStuff", ConfigValueFactory.fromAnyRef(true));
    .withValue("plugin.doMoreStuff", ConfigValueFactory.fromAnyRef(false));

config.save(true);
```